script/DeployKarma.s.sol

```diff
@@ -4,17 +4,25 @@ pragma solidity ^0.8.26;
 import { BaseScript } from "./Base.s.sol";
 import { DeploymentConfig } from "./DeploymentConfig.s.sol";
 
+import { TransparentProxy } from "../src/TransparentProxy.sol";
+
 import { Karma } from "../src/Karma.sol";
 
 contract DeployKarmaScript is BaseScript {
-    function run() public returns (Karma) {
+    function run() public returns (Karma, DeploymentConfig) {
         DeploymentConfig deploymentConfig = new DeploymentConfig(broadcaster);
         (address deployer,,) = deploymentConfig.activeNetworkConfig();
 
         vm.startBroadcast(deployer);
-        address karma = address(new Karma());
+
+        // Deploy Karma logic contract
+        bytes memory initializeData = abi.encodeCall(Karma.initialize, (deployer));
+        address impl = address(new Karma());
+        // Create upgradeable proxy
+        address proxy = address(new TransparentProxy(impl, initializeData));
```

Comment:
I thinks this one here can just be an `ERC1967Proxy` as it doesn't need to expose the `implementation()` function